addons/binding/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/handler/FroniusHandler.java

```diff
@@ -0,0 +1,369 @@
+                ValueUnit day = inverterRealtimeResponse.getBody().getData().getDayEnergy();
+                day.setUnit("kWh");
+                return day;
+            case FroniusBindingConstants.InverterDataChannelPac:
+                return inverterRealtimeResponse.getBody().getData().getPac();
+            case FroniusBindingConstants.InverterDataChannelTotal:
+                ValueUnit total = inverterRealtimeResponse.getBody().getData().getTotalEnergy();
+                total.setUnit("MWh");
+                return total;
+            case FroniusBindingConstants.InverterDataChannelYear:
+                ValueUnit year = inverterRealtimeResponse.getBody().getData().getYearEnergy();
+                year.setUnit("MWh");
+                return year;
+            case FroniusBindingConstants.InverterDataChannelFac:
+                return inverterRealtimeResponse.getBody().getData().getFac();
+            case FroniusBindingConstants.InverterDataChannelIac:
+                return inverterRealtimeResponse.getBody().getData().getIac();
+            case FroniusBindingConstants.InverterDataChannelIdc:
+                return inverterRealtimeResponse.getBody().getData().getIdc();
+            case FroniusBindingConstants.InverterDataChannelUac:
+                return inverterRealtimeResponse.getBody().getData().getUac();
+            case FroniusBindingConstants.InverterDataChannelUdc:
+                return inverterRealtimeResponse.getBody().getData().getUdc();
+        }
+
+        if (powerFlowResponse == null) {
+            return null;
+        }
+        switch (fieldName) {
+            case FroniusBindingConstants.PowerFlowpGrid:
+                return powerFlowResponse.getBody().getData().getSite().getPgrid();
+            case FroniusBindingConstants.PowerFlowpLoad:
+                return powerFlowResponse.getBody().getData().getSite().getPload();
+            case FroniusBindingConstants.PowerFlowpAkku:
+                return powerFlowResponse.getBody().getData().getSite().getPakku();
+        }
+
+        return null;
+    }
+
+    /**
+     * Get new data
+     *
+     */
+    private void updateData() {
+        FroniusConfiguration config = getConfigAs(FroniusConfiguration.class);
+        inverterRealtimeResponse = getRealtimeData(config.ip, config.deviceId);
+        powerFlowResponse = getPowerFlowRealtime(config.ip);
+    }
+
+    /**
+     * Make the PowerFlowRealtimeDataRequest
+     *
+     * @param ip
+     * @return {PowerFlowRealtimeResponse}
+     */
+    private PowerFlowRealtimeResponse getPowerFlowRealtime(String ip) {
+        PowerFlowRealtimeResponse result = null;
+        boolean resultOk = false;
+        String errorMsg = null;
+
+        try {
+
+            String location = PowerFlowRealtimeData.replace("%IP%", StringUtils.trimToEmpty(ip));
+            logger.debug("URL = {}", location);
+            URL url = new URL(location);
```

Comment:
Instead of using Javas URLConnection, please use our [HttpUtil](https://github.com/eclipse/smarthome/blob/master/bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpUtil.java#L77) which uses jetty to retrieve the data.